plugin.video.embycon/resources/lib/downloadutils.py

```diff
@@ -0,0 +1,401 @@
+                conn = httplib.HTTPSConnection(server, timeout=40, context=ssl._create_unverified_context())
+            else:
+                log.debug("Connection: HTTP")
+                conn = httplib.HTTPConnection(server, timeout=40)
+
+            head = self.getAuthHeader(authenticate)
+            log.debug("HEADERS : " + str(head))
+
+            if (postBody != None):
+                if isinstance(postBody, dict):
+                    content_type = "application/json"
+                    postBody = json.dumps(postBody)
+                else:
+                    content_type = "application/x-www-form-urlencoded"
+
+                head["Content-Type"] = content_type
+                log.debug("Content-Type : " + content_type)
+
+                log.debug("POST DATA : " + postBody)
+                conn.request(method=method, url=urlPath, body=postBody, headers=head)
+            else:
+                conn.request(method=method, url=urlPath, headers=head)
+
+            data = conn.getresponse()
+            log.debug("GET URL HEADERS : " + str(data.getheaders()))
+
+            if int(data.status) == 200:
+                retData = data.read()
+                contentType = data.getheader('content-encoding')
+                log.debug("Data Len Before : " + str(len(retData)))
+                if (contentType == "gzip"):
+                    retData = StringIO.StringIO(retData)
+                    gzipper = gzip.GzipFile(fileobj=retData)
+                    return_data = gzipper.read()
+                else:
+                    return_data = retData
+                log.debug("Data Len After : " + str(len(return_data)))
+                log.debug("====== 200 returned =======")
+                log.debug("Content-Type : " + str(contentType))
+                log.debug(return_data)
+                log.debug("====== 200 finished ======")
+
+            #elif (int(data.status) == 301) or (int(data.status) == 302):
+            #    try:
+            #        conn.close()
+            #    except:
+            #        pass
+            #    return data.getheader('Location')
+
+            elif int(data.status) >= 400:
+                error = "HTTP response error: " + str(data.status) + " " + str(data.reason)
+                log.error(error)
+                if suppress is False:
+                    if popup == 0:
+                        xbmc.executebuiltin("Notification(%s, %s)" % (self.addon_name, i18n('url_error_') % str(data.reason)))
+                    else:
+                        xbmcgui.Dialog().ok(self.addon_name, i18n('url_error_') % str(data.reason))
+                log.error(error)
+
+        except Exception, msg:
+            error = "Unable to connect to " + str(server) + " : " + str(msg)
+            log.error(error)
+            if suppress is False:
+                if popup == 0:
+                    xbmc.executebuiltin("Notification(%s, %s)" % (self.addon_name, i18n('url_error_') % str(msg)))
```

Comment:
you can use `xbmcgui.Dialog().notification()`. It is always better to use the python API's rather than builtin functions